### PR TITLE
Unified deployment for the collator node milestone delivery

### PR DESCRIPTION
**Pull request by markblaize**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/1285

### CODE REVIEW

File: deliveries/unified_collator_node_deployment.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+* **Milestone Number:** 1
+
+**Context** (optional)
+
+
+**Deliverables**
+| Number | Deliverable | Link | Notes |
+| ------ | ----------- | ---- | ----- |
+| **0a.** | License | https://github.com/for-parity-with-love/unified-collator-node-deployment/blob/master/LICENSE | Apache 2.0 |
+| **0b.** | Documentation | [Project introduction](https://github.com/for-parity-with-love/unified-collator-node-deployment/blob/master/README.md); [AWS Deployment](https://github.com/for-parity-with-love/unified-collator-node-deployment/blob/master/AWS/README.md); [GCP Deployment](https://github.com/for-parity-with-love/unified-collator-node-deployment/blob/master/GCP/README.md) | Project introduction contains general information about purpose and navigation; AWS\GCP Deployment provides specific instructions to deploy collator to the specific cloud |
```
Comment (semuelle) [inline deliveries/unified_collator_node_deployment.md:16]:
```suggestion
| **0b.** | Documentation | [Project introduction](https://github.com/for-parity-with-love/unified-collator-node-deployment/blob/master/README.md); [AWS Deployment](https://github.com/for-parity-with-love/unified-collator-node-deployment/blob/master/AWS/README.MD); [GCP Deployment](https://github.com/for-parity-with-love/unified-collator-node-deployment/blob/master/GCP/README.MD) | Project introduction contains general information about purpose and navigation; AWS\GCP Deployment provides specific instructions to deploy collator to the specific cloud |
```

**Comment by semuelle** [comment]
Hi Mark. I am trying to run the `deploy.sh` script in your AWS examples folder. I installed an configured AWS CLI, but I get the following error message. Would I have to modify [either of these lines](https://github.com/for-parity-with-love/unified-collator-node-deployment/blob/32c5ee536674ac290b7ca4b596c2893b8e933086/AWS/examples/tfvars/astar.tfvars#L2-L4)?

```sh
❯ ./deploy.sh
1) astar
2) moonbeam
3) karura
4) Quit
Please select desired collator to deploy: 1
astar was chosen...
making setup...
terraform apply -var-file examples/tfvars/astar.tfvars
2023-03-07T16:28:06.703+0100 [ERROR] provider.terraform-provider-aws_v4.48.0_x5: Response contains error diagnostic: tf_req_id=9155014e-4c1a-6187-6b61-08a2c3ce59e6 @caller=github.com/hashicorp/terraform-plugin-go@v0.14.2/tfprotov5/internal/diag/diagnostics.go:55 diagnostic_severity=ERROR tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/hashicorp/aws tf_rpc=Configure @module=sdk.proto diagnostic_detail= diagnostic_summary="configuring Terraform AWS Provider: failed to get shared config profile, collator" timestamp=2023-03-07T16:28:06.703+0100
2023-03-07T16:28:06.703+0100 [ERROR] vertex "provider[\"registry.terraform.io/hashicorp/aws\"]" error: configuring Terraform AWS Provider: failed to get shared config profile, collator
╷
│ Error: configuring Terraform AWS Provider: failed to get shared config profile, collator
│ 
│   with provider["registry.terraform.io/hashicorp/aws"],
│   on provider.tf line 1, in provider "aws":
│    1: provider "aws" {
│ 
╵
terraform apply -var-file examples/tfvars/astar.tfvars
```

**Review by semuelle — CHANGES_REQUESTED** [review]
Some more notes:

- According to the contract, deliverable `0b` should contain a "step-by-step guide on how to [...] maintain and monitor the collator node". 
- Also, `0c` should contain "comprehensive infrastructure tests". The deployment script and example configurations are quite useful, but I would like to ask you to document these (add a README to the `examples` folders as well as inline docs) so that they become less dense and easier to maintain/modify.
- We don't usually approve deliveries unless all deliverables are included. It'd be great if you could add `0d` asap.

**Comment by semuelle** [comment]
pinging @markblaize

**Comment by markblaize** [comment]
> Hi Mark. I am trying to run the `deploy.sh` script in your AWS examples folder. I installed an configured AWS CLI, but I get the following error message. Would I have to modify [either of these lines](https://github.com/for-parity-with-love/unified-collator-node-deployment/blob/32c5ee536674ac290b7ca4b596c2893b8e933086/AWS/examples/tfvars/astar.tfvars#L2-L4)?
> 
> ```shell
> ❯ ./deploy.sh
> 1) astar
> 2) moonbeam
> 3) karura
> 4) Quit
> Please select desired collator to deploy: 1
> astar was chosen...
> making setup...
> terraform apply -var-file examples/tfvars/astar.tfvars
> 2023-03-07T16:28:06.703+0100 [ERROR] provider.terraform-provider-aws_v4.48.0_x5: Response contains error diagnostic: tf_req_id=9155014e-4c1a-6187-6b61-08a2c3ce59e6 @caller=github.com/hashicorp/terraform-plugin-go@v0.14.2/tfprotov5/internal/diag/diagnostics.go:55 diagnostic_severity=ERROR tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/hashicorp/aws tf_rpc=Configure @module=sdk.proto diagnostic_detail= diagnostic_summary="configuring Terraform AWS Provider: failed to get shared config profile, collator" timestamp=2023-03-07T16:28:06.703+0100
> 2023-03-07T16:28:06.703+0100 [ERROR] vertex "provider[\"registry.terraform.io/hashicorp/aws\"]" error: configuring Terraform AWS Provider: failed to get shared config profile, collator
> ╷
> │ Error: configuring Terraform AWS Provider: failed to get shared config profile, collator
> │ 
> │   with provider["registry.terraform.io/hashicorp/aws"],
> │   on provider.tf line 1, in provider "aws":
> │    1: provider "aws" {
> │ 
> ╵
> terraform apply -var-file examples/tfvars/astar.tfvars
> ```

sorry for the delay here, I've added a point to the examples readme about the configuration of an additional AWS profile (usually located here ~/.aws/credentials)
```
[collator]
aws_access_key_id = <aws_access_key_id>
aws_secret_access_key = <aws_secret_access_key>
```

for simplicity you can just copy your default profile and name it "collator"

**Comment by markblaize** [comment]
> Some more notes:
> 
> * According to the contract, deliverable `0b` should contain a "step-by-step guide on how to [...] maintain and monitor the collator node".
> * Also, `0c` should contain "comprehensive infrastructure tests". The deployment script and example configurations are quite useful, but I would like to ask you to document these (add a README to the `examples` folders as well as inline docs) so that they become less dense and easier to maintain/modify.
> * We don't usually approve deliveries unless all deliverables are included. It'd be great if you could add `0d` asap.

Sorry for the delay once again,
>We don't usually approve deliveries unless all deliverables are included. It'd be great if you could add 0d asap.
We have sent it for review and @Stefan Orlić gave us a few comments that we fixing and slightly rewriting an article

>Also, 0c should contain "comprehensive infrastructure tests". The deployment script and example configurations are quite useful, but I would like to ask you to document these (add a README to the examples folders as well as inline docs) so that they become less dense and easier to maintain/modify.
I've updated the documentation, please check

>According to the contract, deliverable 0b should contain a "step-by-step guide on how to [...] maintain and monitor the collator node".
We will provide more details on which maintenance actions are supported within our project and more details on monitoring except for telemetry web UI.

**Comment by semuelle** [comment]
Thanks for the quick reply, @markblaize. Please ping me here once you're done.

**Comment by markblaize** [comment]
> Thanks for the quick reply, @markblaize. Please ping me here once you're done.

Hi @semuelle could you please check this updated readme

https://github.com/for-parity-with-love/unified-collator-node-deployment/blob/master/AWS/README.MD#maintenance

**Review by semuelle — APPROVED** [review]
Thanks for the update, @markblaize. Your milestone is hereby approved, congratulations! You can find my evaluation notes [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/unified-collator-deployment_1_semuelle.md).

**Comment by RouvenP** [comment]
hi @AnnaYShch we transferred the payment today.

**Comment by markblaize** [comment]
> hi @AnnaYShch we transferred the payment today.

Thank you once again, guys! It was a pleasure to create something for such a big and long-lasting blockchain.

We look forward to hearing the first comments from the community about the project as well as investigating where our expertise could be useful within your ecosystem

**Pull request merged**